File: CS/PrintCellStyleSelector/Classes.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Data;
using DevExpress.Xpf.Grid;

namespace PrintCellStyleSelector
{
    public class Data
    {
        public static List<Customer> GetData()
        {
                List<Customer> list = new List<Customer>();
                for (int i = 0; i < 10; i++)
                {
                    list.Add(new Customer()
                    {
                       ID=i,
                        FirstName="FirstName"+i,
                        LastName="LastName"+i,
                        Birthday=new DateTime(1980+i,i+1,10+i)
                    });
                }
                return list;
        }
    }

    public class Customer
    {
        public int ID{get;set;}
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public DateTime Birthday { get; set; }
    }

    public class Converter : IMultiValueConverter
    {
        public object Convert(object[] values, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            return values.ToList<object>(); ;
        }

        public object[] ConvertBack(object value, Type[] targetTypes, object parameter, System.Globalization.CultureInfo culture)
        {
            return new object[0];
        }
    }

    public class TemplateSelector : DataTemplateSelector
    {
        public ResourceDictionary Dictionary { get; set; }

        public override DataTemplate SelectTemplate(object item, DependencyObject container)
        {
            List<object> values = item as List<object>;
            int n = 0;
            Math.DivRem(Math.Abs((int) values[0]),2,out n);
            if ( n>= 1)
                return Dictionary["DefaultPrintTextEdit"] as DataTemplate;
            else
                return Dictionary["CustomPrintTextEdit"] as DataTemplate;
        }
    }
}

File: VB/PrintCellStyleSelector/Classes.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Data
Imports DevExpress.Xpf.Grid

Namespace PrintCellStyleSelector
    Public Class Data
        Public Shared Function GetData() As List(Of Customer)
                Dim list As New List(Of Customer)()
                For i As Integer = 0 To 9
                    list.Add(New Customer() With {.ID=i, .FirstName="FirstName" & i, .LastName="LastName" & i, .Birthday = New Date(1980+i,i+1,10+i)})
                Next i
                Return list
        End Function
    End Class

    Public Class Customer
        Public Property ID() As Integer
        Public Property FirstName() As String
        Public Property LastName() As String
        Public Property Birthday() As Date
    End Class

    Public Class Converter
        Implements IMultiValueConverter

        Public Function Convert(ByVal values() As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IMultiValueConverter.Convert
            Return values.ToList()

        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetTypes() As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object() Implements IMultiValueConverter.ConvertBack
            Return New Object(){}
        End Function
    End Class

    Public Class TemplateSelector
        Inherits DataTemplateSelector

        Public Property Dictionary() As ResourceDictionary

        Public Overrides Function SelectTemplate(ByVal item As Object, ByVal container As DependencyObject) As DataTemplate
            Dim values As List(Of Object) = TryCast(item, List(Of Object))
            Dim n As Integer = 0
            Math.DivRem(Math.Abs(DirectCast(values(0), Integer)),2,n)
            If n>= 1 Then
                Return TryCast(Dictionary("DefaultPrintTextEdit"), DataTemplate)
            Else
                Return TryCast(Dictionary("CustomPrintTextEdit"), DataTemplate)
            End If
        End Function
    End Class
End Namespace
